templates/highmemagent-deployment.yaml

```
@@ -94,6 +94,8 @@ spec:
               mountPath: /opt/buildagent/temp
             - name: agent-work
               mountPath: /opt/buildagent/work
+      imagePullSecrets:
+{{ toYaml .Values.highmemagent.imagePullSecrets | indent 8 }}
```

The better way of doing this is by wrapping this in an `if-end` so that this section won't be rendered if there is no `imagePullSecret`.

---

This is an array. If the array is empty, nothing happens. What's the purpose to add complexity if it doesn't change anything?

---

the value is that with your version, you always render the `imagePullSecret` yaml tag, even when there is none.

also, after looking it up, there seems to be this way of doing this:

```suggestion
{{- include "highmemagent.imagePullSecrets" . | indent 6 }}
```

source: https://github.com/helm/charts/blob/master/stable/postgresql/templates/statefulset.yaml#L50